rust/parquet/src/data_type.rs

```diff
@@ -23,6 +23,8 @@ use std::mem;
 use byteorder::{BigEndian, ByteOrder};
 
 use crate::basic::Type;
+use crate::column::reader::{ColumnReader, ColumnReaderImpl};
```

Comment:
That could result in a circular dependency, we should not be doing that, or we should at least factor out the interfaces and structs there.

Reply:
Rust handles circular, inter crate dependencies fine 🤷‍♂

Reply:
It still does not mean we should be doing it.